### DISCUSSION TITLE
Retrieving the full link from old shortened git.io links

### DISCUSSION THREAD

**Discussion by swaroopg92** [community/community#78518] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi everyone,
I am trying to find out if I can request Git Hub or if there is a way to expand or retrieve the Git.io URL.

I know that git.io is deprecated. I also see the information in the announcement: "Effective Friday, April 29, 2022 all links on git.io will stop redirecting."

I found many old links I created for my GitHub pages that I didn't copy in time, but now they don't work.
Is there any way that I can access that?
Again, I don't want the git.io link to work, but retrieve the long URL stored in the shortened URL.

One example shortened link: https://git.io/JuAGl, I want to retrieve the full link?

The Google search and forum search only talk about Git not accepting new links and the service being deprecated.

Any help is appreciated.
Thanks

**Comment by glasser** [comment]
Yes, it would be lovely if it were possible to do a one-time batch lookup of a set of links.

**Comment by Gecko51** [comment]
Short answer: GitHub fully decommissioned the git.io redirect map when the service was sunset, so there's no API or endpoint left to resolve a short link to its original target. The mapping table is gone. You can't ask GitHub to look them up anymore.

That said, you've got a few realistic recovery paths depending on what the links pointed to.

If the links pointed to a GitHub repo, gist, issue, PR, or Pages site you own:

- Check the Wayback Machine: `https://web.archive.org/web/*/git.io/JuAGl`. If the short link was ever crawled (and a lot were), the archive will show the redirect target. This is by far the highest hit rate I've seen for old git.io links.
- Try archive.today: `https://archive.ph/git.io/JuAGl`. Sometimes catches things Wayback missed.
- Plain Google/Bing search with the short URL in quotes. README files, docs, blog posts, and Stack Overflow threads often mentioned both the short and long URLs side by side.

If you know what the original target probably looked like:

- GitHub's own URL patterns are pretty guessable. Pages sites are `<user>.github.io/<repo>`, raw content is `raw.githubusercontent.com/<user>/<repo>/<ref>/<path>`, etc. If you remember the general repo it pointed to, reconstructing the target manually is often faster than any recovery attempt.
- Check the commit history, issues, and PR descriptions in the relevant repo. You or others often referenced the full URL before you shortened it.

For batch recovery:

- If you have a list of the short links, loop them through Wayback's availability API:

```bash
curl -s "https://archive.org/wayback/available?url=git.io/JuAGl" | jq '.archived_snapshots.closest.url'
```

That gives you the earliest archived snapshot, which usually resolved the redirect.

If none of those hit, that specific mapping is likely lost. GitHub was clear that they weren't publishing the decommissioned database, and I haven't seen any third-party mirror of it floating around.
